packages/stark-ui/src/modules/minimap/components/minimap.component.ts

```diff
@@ -0,0 +1,173 @@
+ *
+ * @param string[] - visibleItems Array of names of the items that are visible.
+ *
+ * @param showHideItem EventEmitter to be called when call showHideItem callback.
+ *
+ */
+@Component({
+	selector: "stark-minimap",
+	templateUrl: "./minimap.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkMinimapComponent implements OnInit, OnChanges, OnDestroy {
+	@Input()
+	public mode?: StarkMinimapComponentMode;
+	@Input()
+	public parentItems: StarkMinimapItemProperties[];
+	@Input()
+	public parentVisibleItems: string[];
+
+	@Output()
+	public showHideItem: EventEmitter<StarkMinimapItemProperties> = new EventEmitter<StarkMinimapItemProperties>();
+
+	public items: StarkMinimapItemProperties[];
+	public visibleItems: string[];
+
+	public windowClickHandler: EventListener;
+	public isDisplayedMenu: boolean;
+
+	public constructor(@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService, private ngZoneService: NgZone) {}
+
+	public ngOnInit(): void {
+		this.visibleItems = _cloneDeep(this.parentVisibleItems) || [];
+		this.items = _cloneDeep(this.parentItems) || [];
+
+		this.isDisplayedMenu = false;
+		this.attachWindowClickHandler();
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnChanges(changesObj: SimpleChanges): void {
+		if (changesObj["parentItems"] && !_isEqual(this.parentItems, changesObj["parentItems"].previousValue)) {
+			this.logger.debug(componentName + ": items changed... Recreating items config...", changesObj["parentItems"]);
+			this.items = _cloneDeep(this.parentItems);
+		}
+
+		if (changesObj["parentVisibleItems"] && !_isEqual(this.parentVisibleItems, changesObj["parentVisibleItems"].previousValue)) {
+			this.logger.debug(
+				componentName + ": visibleItems changed... Recreating visibleItems config...",
+				changesObj["parentVisibleItems"]
+			);
+			this.visibleItems = _cloneDeep(this.parentVisibleItems);
+		}
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnDestroy(): void {
+		this.logger.debug(componentName + ": removing clickHandler");
+		this.removeWindowClickHandler();
+	}
+
+	/**
+	 * Retrieve the label of an item
+	 * @param StarkMinimapItemProperties - the item whose label we want to retrieve
+	 * @returns string - the label of the item
+	 */
+	public getItemLabel(item: StarkMinimapItemProperties): string {
+		if (item.label) {
+			return item.label;
+		}
+
+		return item.name;
+	}
+
+	/**
+	 * Attach a handler when a click is performed on the window
+	 */
+	public attachWindowClickHandler(): void {
+		this.windowClickHandler = (event: Event) => {
+			if (this.isDisplayedMenu) {
+				const parentElement: Element | undefined = StarkDOMUtil.searchParentElementByClass(
+					<Element>event.target,
+					"stark-minimap-dropdown-toggle"
+				);
+
+				if (!parentElement) {
+					this.ngZoneService.run(() => this.toggleDropdownMenu());
```

Comment:
I don't think we need to use the `ngZoneService` itself, perhaps we can achieve the same by using the [ApplicationRef](https://angular.io/api/core/ApplicationRef) instead inside the `toggleDropdownMenu()` method